great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -861,14 +876,14 @@ def _generate_rule_overrides_from_batch_request(
 
         for rule in rules:
             domain_builder = rule.domain_builder
-            if isinstance(domain_builder, ColumnDomainBuilder):
-                domain_builder.batch_request = copy.deepcopy(batch_request)
+            if domain_builder.domain_type == MetricDomainTypes.COLUMN:
+                domain_builder.batch_request = safe_deep_copy(batch_request)
```

Comment:
@cdkini If we must copy a `batch_request`, then we should simply use `copy.deepcopy()` (because the internal implementation of `BatchRequest` already overrides `__deepcopy__()`.  However, I feel that we should not be copying it, because the users here do not modify any part of it.  If you can, please see if tests pass without copying.  Thanks.